src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -223,6 +232,11 @@ private HttpProtocols SelectProtocol()
                 return HttpProtocols.Http1;
             }
 
+            if (_context.Protocols == HttpProtocols.Http3)
```

Comment:
Nit: If this is never true when `(_context.Protocols & HttpProtocols.Http1) || (_context.Protocols & HttpProtocols.Http2)`, I'd just move this to the very top of the method.